### PR TITLE
Feat: add run-all support

### LINKED ISSUE #438: Add run-all support
We've added the ability to run `terragrunt run-all`, which runs multiple modules simultaneously.

The structure of the output result has been changed (**only** when `isTerragruntRunAll` is provided) to be:
```json
[{
   "moduleName": "<some name>"
   "output": { }
}]
```

This is noted by the `isTerragruntRunAll` field on the blueprint / template.
The blueprint / template's `type` MUST be `terragrunt`, and the terragrunt `version` MUST be `0.28.1` or above.

[API Reference](https://env0.stoplight.io/docs/api/2b138939cbeec-create-template)

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
resolves #438 

### Solution
1. added the new boolean flag.
2. added unit tests.
3. added semver to go.mod for comparing versions.

### CODE REVIEW

File: client/template.go [hunk 1]
```diff
@@ -125,6 +129,21 @@ func (payload TemplateCreatePayload) Validate() error {
 		return errors.New("must supply terragrunt version")
 	}
 
+	if payload.IsTerragruntRunAll {
+		if payload.Type != "terragrunt" {
+			return errors.New(`can't set is_terragrunt_run_all to "true" for non-terragrunt template`)
+		}
+
+		c, _ := semver.NewConstraint(">= 0.28.1")
```
Comment (TomerHeber) [inline client/template.go:137]: I added semver library to compare versions and make sure the terragrunt version is above or equal 0.28.1

File: env0/resource_template_test.go [hunk 2]
```diff
@@ -974,4 +982,24 @@ func TestUnitTemplateResource(t *testing.T) {
 
 		runUnitTest(t, testCase, func(mock *client.MockApiClientInterface) {})
 	})
+
+	t.Run("run all with outdated terragrunt version", func(t *testing.T) {
```
Comment (TomerHeber) [inline env0/resource_template_test.go:986]: This verifies the version comparision use-case (when lower than).

File: env0/resource_template_test.go [hunk 3]
```diff
@@ -60,6 +60,7 @@ func TestUnitTemplateResource(t *testing.T) {
 		TerragruntVersion:  "0.35.1",
 		IsGitlabEnterprise: true,
 		TerraformVersion:   "0.15.1",
+		IsTerragruntRunAll: true,
```
Comment (TomerHeber) [inline env0/resource_template_test.go:63]: Updated the acceptance tests to check IsTerragruntRunAll.

File: env0/templates.go [hunk 4]
```diff
@@ -208,6 +208,12 @@ func getTemplateSchema(templateType TemplateType) map[string]*schema.Schema {
 			Description: "the cloudformation file name. Required if the template type is cloudformation",
 			Optional:    true,
 		},
+		"is_terragrunt_run_all": {
```
Comment (TomerHeber) [inline env0/templates.go:211]: This is all that's required. The readResourceData and writeResourceData do all the work automatically.

File: env0/resource_template_test.go (hunk 3 above)
Comment (sagilaufer1992) [reply env0/resource_template_test.go:63]: so now we don't have a non-run-all check? shouldn't it be added on top of a low version of terragrunt that should run as usual, a higher version (>=0.28.1) that isn't run-all that should work, the added low version that is run-all and should fail? total 4 cases (low-all, high-all, low-not-all, high-not-all). that seems to replace one case with another, without adding the counterpart.

**Review by sagilaufer1992 — COMMENTED** [review]
Looks good to me. I think there is a missing test-case, please read my comment and make sure all the cases exist as I suppose they should.

File: env0/resource_template_test.go (hunk 3 above)
Comment (TomerHeber) [reply env0/resource_template_test.go:63]: @sagilaufer1992 - by default IsTerragruntRunAll is false. Therefore, there are some testcases that run with that mode.
I can change one of them to a lower version terragrunt. That should cover all 4 use-cases.

Comment (sagilaufer1992) [reply env0/resource_template_test.go:63]: as we check versions as well, and testing shouldn't take long (unit testing, right?) I'd rather we'll check all 4 use-cases.

Comment (TomerHeber) [reply env0/resource_template_test.go:63]: 8c55b175bd3f06253fd8909710e9d8ee39dca700

Comment (TomerHeber) [reply env0/resource_template_test.go:63]: All 4 use-cases are tested.

gleeUpdatedTemplate - high-all
gitlabUpdatedTemplate - low-not-all
bitbucketUpdatedTemplate - high-not-all
"run all with outdated terragrunt version" - low-all (which is expected to fail).

Comment (TomerHeber) [reply env0/resource_template_test.go:63]: @sagilaufer1992

Comment (sagilaufer1992) [reply env0/resource_template_test.go:63]: was about to comment that I found all, and thank you. so extra thank you for making it easier for others as well 🙏

**Review by sagilaufer1992 — APPROVED** [review]
great job 👍

**Pull request merged**